src/reducers/linkages.js

```diff
@@ -5,26 +5,25 @@ import {
   REMOVE_LINKAGE,
 } from 'actions';
 
+import { dissoc, path, assoc } from 'ramda';
+
+const getLinkagesFromAction = action =>
+  path(['payload', 'state', 'network', 'linkages'], action);
+
 export default (state = {}, action) => {
   switch (action.type) {
     case NEW_NETWORK:
       return {};
-    case ADD_LINKAGE:
-      return {
-        ...state,
-        [Date.now()]: action.payload.linkage,
-      };
+    case ADD_LINKAGE: {
+      const { linkage } = action.payload;
+      return assoc(Date.now(), linkage, state);
+    }
     case REMOVE_LINKAGE: {
       const { id } = action.payload;
-      const newState = { ...state };
-      delete newState[id];
-
-      return newState;
-    }
-    case LOAD_NETWORK: {
-      const { linkages } = action.payload.state.network;
-      return linkages || {};
+      return dissoc(id, state);
     }
+    case LOAD_NETWORK:
+      return getLinkagesFromAction(action) || {};
```

Comment:
You can use `defaultTo` here and [pipe](https://ramdajs.com/docs/#pipe) to combine these function:

```js
const getLinkagesFromAction = 
  path(['payload', 'state', 'network', 'linkages']);

// you can choose a better name if you wanna...
const getLinkagesOrDefaultFromAction = pipe(getLinkagesFromAction, defaultTo({}))
...
return getLinkagesOrDefaultFromAction(action);
```

Just to be clear, I can rewrite this function like that:
```js
const getLinkagesFromAction = 
  path(['payload', 'state', 'network', 'linkages']);
const defaultToEmptyObject = defaultTo({})

const getLinkagesOrDefaultFromAction = action => {
  const linkages = getLinkagesFromAction(action);
  
  return defaultToEmptyObject(linkages)
}
```
😄